test/testSimplifier.ts

```diff
@@ -171,11 +171,23 @@ function tokenizeKeySequence(sequence: string): string[] {
   let key = '';
   let result: string[] = [];
 
+  // no close bracket, probably trying to do a left shift, take literal
```

Comment:
I tried walking through this a couple of times, it's probably cause it's the end of the day, but I can't grok this code. 

The code might be simpler if instead we walk the string, if we see `<`, find the matching `>`. If there's no matching `<`, then it's a left shift.

Reply:
So what I'm doing is sort of the opposite, I'm looking for another `<` following a `<`, and then working back instead of looking ahead.

 If I see a second `<` (i.e. `isBracketedKey` is already truthy when I hit a `<`) I know whatever was previous was a left shift and subsequent literal character inputs, so I take the currently built `key` and tokenize it one character per token.

After the tokenizing loop has run, if `isBracketedKey` is truthy I know we also saw an unclosed `<` so again I tokenize the currently built `key` as one character per token.

I don't mind looking ahead for closing `>` instead if you still prefer that. I didn't do that because it's a bit more complicated in the case of `keysPressed` like `<<<ESC>` where I would have to start counting open `<`s and looking for matches or recursing which seemed overcomplicated for a convenience test helper function. That would be starting to get into serious parsing territory.

Reply:
Although now that I've explained this I've found a bug, I will update.

Reply:
Ah, thanks for the explanation.